### PR TITLE
Implement can-do-crypto for TS and mbed-crypto providers

### PR DESCRIPTION
**Pull request by anta5010**
Signed-off-by: Anton Antonov <Anton.Antonov@arm.com>

### CODE REVIEW

**Comment by ionut-arm** [comment]
For the fuzz framework failure I think you just need to run a `cargo update --package psa-crypto` in the `fuzz/` directory as well, as it has a `Cargo.lock` file (it produces and runs a binary).

**Review by ionut-arm — APPROVED** [review]
Looks good, couple of minor comments

File: src/providers/trusted_service/capability_discovery.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            Type::RsaPublicKey | Type::EccPublicKey { .. } | Type::DhPublicKey { .. } => {
+                Ok(can_do_crypto::Result)
+            }
+            Type::RsaKeyPair
+            | Type::EccKeyPair { .. }
+            | Type::DhKeyPair { .. }
+            | Type::RawData
+            | Type::Aes
+            | Type::Camellia
+            | Type::Chacha20 => Ok(can_do_crypto::Result),
```
[the remark this answers is not in the document]
Comment (anta5010) [reply src/providers/trusted_service/capability_discovery.rs:84]: Yes.

File: src/providers/mbed_crypto/capability_discovery.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+use parsec_interface::requests::ResponseStatus::PsaErrorNotSupported;
+use parsec_interface::requests::Result;
+
+impl CanDoCrypto for Provider {
+    fn can_do_crypto_internal(
+        &self,
+        _app_name: ApplicationName,
+        op: can_do_crypto::Operation,
+    ) -> Result<can_do_crypto::Result> {
+        trace!("can_do_crypto_internal for mbed-crypto provider");
```
[the remark this answers is not in the document]
Comment (anta5010) [reply src/providers/mbed_crypto/capability_discovery.rs:19]: I does include indeed. Fixed for all providers

**Pull request merged**